File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d: dict[int, int] = dict()
        for i in range(len(nums)):
            complement: int = target - nums[i]
            if complement in d:
                return [d[complement], i]
            d[nums[i]] = i
        return []
            

File: 1-two-sum/two-sum.cpp
#pragma GCC optimize("O2")

#define fast_io                   \
    ios_base::sync_with_stdio(0); \
    cin.tie(0);                   \
    cout.tie(0);

class Solution 
{
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        fast_io;
        unordered_map<int, int> indices;
        for (int i = 0; i < nums.size(); ++i)
        {
            int complement = target - nums[i];
            if (indices.contains(complement))
                return {indices[complement], i};
            indices[nums[i]] = i;
        }
        return {};
    }
};